### PR TITLE
switch freehand selection map tool to click-click

### PR DESCRIPTION
**Pull request by 3nids**
to be coherent with other map tools
add-ons: possible map move/zoom while selecting, and operation canceling (right-click or ESC)

### CODE REVIEW

**Comment by borysiasty** [comment]
Sorry for a stupid question but I can't imagine the click-click freehand selection. Won't it be a duplicate of the existing Select by Polygon?

**Comment by 3nids** [comment]
> Sorry for a stupid question but I can't imagine the click-click freehand selection. Won't it be a duplicate of the existing Select by Polygon?

well you don't click on the intermediate points. see uploaded demo

**Comment by borysiasty** [comment]
Ah, just click-click for start-stop. I told the question is stupid! :)

**Comment by nyalldawson** [comment]
+1

File: src/app/qgsmaptoolselectfreehand.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ email                : jpalmer at linz dot govt dot nz
 
 QgsMapToolSelectFreehand::QgsMapToolSelectFreehand( QgsMapCanvas *canvas )
   : QgsMapTool( canvas )
-  , mDragging( false )
+  , mActive( false )
```
Comment (nyalldawson) [inline src/app/qgsmaptoolselectfreehand.cpp]: initialize in header

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**